src/Microsoft.TemplateEngine.Cli/New3Command.cs

```diff
@@ -828,6 +836,72 @@ private void ShowTemplateHelp()
             ParameterHelp(allParams, additionalInfo, _hostSpecificTemplateData.HiddenParameterNames);
         }
 
+        private void ShowTemplateNameMismatchHelp()
+        {
+            if (_matchedTemplates.Count == 0)
+            {
+                Reporter.Error.WriteLine("No installed template's names were similiar enough.");
+                return;
+            }
+
+            IDictionary<string, IFilteredTemplateInfo> contextProblemMatches = new Dictionary<string, IFilteredTemplateInfo>();
+            IDictionary<string, IFilteredTemplateInfo> remainingPartialMatches = new Dictionary<string, IFilteredTemplateInfo>();
+
+            // this filtering / grouping ignores language differences.
+            foreach (IFilteredTemplateInfo template in _matchedTemplates)
+            {
+                if (contextProblemMatches.ContainsKey(template.Info.Name) || remainingPartialMatches.ContainsKey(template.Info.Name))
+                {
+                    continue;
+                }
+
+                if (template.MatchDisposition.Any(x => x.Location == MatchLocation.Context && x.Kind != MatchKind.Exact))
+                {
+                    contextProblemMatches.Add(template.Info.Name, template);
+                }
+                else if(template.MatchDisposition.Any(t => t.Location != MatchLocation.Context && t.Kind != MatchKind.Mismatch && t.Kind != MatchKind.Unspecified))
+                {
+                    remainingPartialMatches.Add(template.Info.Name, template);
+                }
+            }
+
+            if (contextProblemMatches.Keys.Count + remainingPartialMatches.Keys.Count > 1)
+            {
+                Reporter.Error.WriteLine($"Unable to determine the desired template from the input template name: [{_templateName}]");
+            }
+            else if (contextProblemMatches.Keys.Count + remainingPartialMatches.Keys.Count == 0)
+            {
+                Reporter.Error.WriteLine($"No templates matched the input template name: [{_templateName}]");
+                return;
+            }
+
+            foreach (IFilteredTemplateInfo template in contextProblemMatches.Values)
+            {
+                if (template.Info.Tags != null && template.Info.Tags.TryGetValue("type", out string type))
+                {
+                    if (string.Equals(type, "item"))
+                    {
+                        Reporter.Error.WriteLine($"\t- [{template.Info.Name}] is an item template. By default it's only created in a target location containing a project. Force creation with the -all flag.");
+                    }
+                    else
+                    {   // project template
+                        Reporter.Error.WriteLine($"\t- [{template.Info.Name}] is a project template. By default it's not created in a target location containing a project. Force creation with the -all flag.");
```

Comment:
It's a single dash: "-all" is the short version, the long version is "--show-all"